### ISSUE TITLE
Swipe not working after deleting a cell

### ISSUE THREAD

**Issue by mhawiegeri** [MortimerGoro/MGSwipeTableCell#163]
In our situation we have a delete button which appears as one of the buttons when swiping to the left. When deleting a cell this way, swiping the next cell (which is now at the location of the deleted cell in the tableView) doesn't work properly. Also clicking at the locations in that cell where the buttons would appear triggers the buttons.

Some important aspects:
* We use **reloadData** after deleting a cell, instead of **deleteRowsAtIndexPath**.
* Also the cells we use have different reuseIdentifiers.
* And, when trying to swipe the second time we start the swipe from the area in the cell where the buttons would appear.

It looks like somehow some old overlay of the deleted cell still exists.

Any suggestions/ideas/fixes would be welcome!

**Comment by MortimerGoro** [comment]
Hello, ¿can you send me a testcase? That would be ready helpful because trying to reproduce this things sometimes takes time and I want to focus on fixing it ;)

**Comment by mhawiegeri** [comment]
Of course.

We also tried to get the situation in the MailAppDemo. Let's use this as a starting point.

I changed the following in MailViewController.m:
```
-(void) deleteMail:(NSIndexPath *) indexPath
{
    [demoData removeObjectAtIndex:indexPath.row];
//    [_tableView deleteRowsAtIndexPaths:@[indexPath] withRowAnimation:UITableViewRowAnimationLeft];
    [_tableView reloadData];
}
```
and
```
- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    MailData * data = [demoData objectAtIndex:indexPath.row];
    
    NSString * identifier = data.from;
    MailTableCell * cell = [tableView dequeueReusableCellWithIdentifier:identifier];
    if (data.type == 0) {
        cell = [[MailTableCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:@"MailTableCell"];
    } else {
        cell = [[MailTableTwoCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:@"MailTableTwoCell"];
        cell.backgroundColor = [UIColor greenColor];
    }
    
    cell.delegate = self;
    
    cell.mailFrom.text = data.from;
    cell.mailSubject.text = data.subject;
    cell.mailMessage.text = data.message;
    cell.mailTime.text = data.date;
    [self updateCellIndicactor:data cell:cell];
    return cell;
}
```
Also added `@property (nonatomic, assign) NSInteger type;` to MailData.

Using this you get a similar situation as in our app. 
Then take the next steps:
* Swipe the top cell to the left until the buttons are shown
* Press Trash
* -> Try swiping the new top cell to the left. Start the swipe from the right side of the cell (the area where the buttons would be).
* -> Or, simply tap the new top cell at the location the trash button would be.

Then we get unexpected behaviour. The new top cell somehow thinks it has been swiped already.

**Comment by adrianilie9** [comment]
this bug happens to me too. i found a workaround.
before deleting the cell (or reloading the tableView in your case)

    MGSwipeTableCell *cell = (MGSwipeTableCell *)[self.tableView cellForRowAtIndexPath:selectedItemIndexPath];
    [cell hideSwipeAnimated:YES];

**Comment by MortimerGoro** [comment]
Fixed in 1.5.4 ;)
